core/src/main/java/io/grpc/internal/ServiceConfigUtil.java

```diff
@@ -642,4 +697,64 @@ private static long saturatedAdd(long a, long b) {
     // we did over/under flow, if the sign is negative we should return MAX otherwise MIN
     return Long.MAX_VALUE + ((naiveSum >>> (Long.SIZE - 1)) ^ 1);
   }
+
+  /**
+   * A LoadBalancingConfig that includes the policy name (the key) and its raw config value (parsed
+   * JSON).
+   */
+  public static final class LbConfig {
```

Comment:
package privatae

Reply:
If this will be public for use by LB policies, then it shouldn't be in `internal`. No implementation should be required to use `internal` code. Since hierarchical policies are core to the design, that means we need a non-internal way to handle subpolicies. 

We could expose the parser as a utility function that returns the normal `(cookie, status)`. We would then have a utility (on Helper?) to construct a LB policy provided the cookie. The cookie would be this LbConfig, but LbConfig would be private.

If parent policies are allowed to special-case certain names (which seems like it may be a bad idea, but I expect there are arguments on both sides), then we would be required to make the class public and it would be part of our API in `io.grpc` or say those use-cases must parse the LbConfig themselves (and access the LB registry manually).

(For the gRPC-LB v1 goo, it will special-case certain names, but in that case it isn't actually going to need the pre-built LbConfig parsing since it will be ignoring the subpolicy configuration anyway.)

-----

Note that this wouldn't be used as-is with the API I was talking about. It wouldn't be a `rawConfigValue` but instead the child policy's cookie. But I understand we can't plumb that today.

Reply:
If it shouldn't be `internal`, it might be `@Internal` for the moment.

Reply:
I meant that this **internal** class has to have **public access** as of now, but I expect it to evolve into a public API in a possibly different form. This PR is **not** going to propose any public API. It is merely an internal refactor to keep our own LoadBalancingConfig handling path sane.